tests/test_assets_info.py

```diff
@@ -31,11 +31,13 @@ def test_update(self):
             }
         }
         self.assets_info.update({}, core_stats, "")
+        self.assets_info.update({}, core_stats, "")
```

Comment:
I suppose this line was duplicated by mistake
```suggestion
```

Reply:
Unless it's part of the reason why I didn't quite understand the fix :D

Reply:
It looks like you're calling `.update()` twice, so that the total generated energy is accumulated (as you say in the comment). If this is the case, I would suggest to actually check the value twice. For example:
```python
self.assets_info.update({}, core_stats, "")
expected_res = {
                "number_of_load_type": 3,
                "number_of_pv_type": 2,
                "total_energy_demand_kwh": 0.06,
                "total_energy_generated_kwh": 25
                }
actual_res = {key: self.assets_info.raw_results[key] for key in expected_res.keys()}
assert expected_res == actual_res

# total_energy_demanded_wh is already accumulated value
expected_res = {
                "number_of_load_type": 3,
                "number_of_pv_type": 2,
                "total_energy_demand_kwh": 0.06,
                "total_energy_generated_kwh": 50
                }
actual_res = {key: self.assets_info.raw_results[key] for key in expected_res.keys()}
assert expected_res == actual_res
```

Reply:
Yes, I call `.update()` twice on purpose. You are right, let me check the values twice, too.